src/exports.cc

```diff
@@ -833,16 +833,31 @@ Word wasi_unstable_args_sizes_get(void *raw_context, Word argc_ptr, Word argv_bu
 Word wasi_unstable_clock_time_get(void *raw_context, Word clock_id, uint64_t precision,
                                   Word result_time_uint64_ptr) {
 
-  if (clock_id != 0 /* realtime */) {
+#if !defined(_MSC_VER)
+  clockid_t id = CLOCK_REALTIME;
+  switch (clock_id) {
+  case 0 /* realtime */:
+    break;
+  case 1 /* monotonic */:
+    id = CLOCK_MONOTONIC;
+    break;
+  default:
+    // process_cputime_id and thread_cputime_id are not supported yet.
     return 58; // __WASI_ENOTSUP
   }
-
+  struct timespec tpe;
+  clock_gettime(id, &tpe);
```

Comment:
That's not the best way to implement this. Instead, you should return `Envoy::RealTimeSource::monotonicTime()`,  which requires plumbing this via `Context`.

Reply:
thanks!